go/logic/applier.go

```diff
@@ -30,16 +30,18 @@ const (
 // Applier is the one to actually write row data and apply binlog events onto the ghost table.
 // It is where the ghost & changelog tables get created. It is where the cut-over phase happens.
 type Applier struct {
-	connectionConfig *mysql.ConnectionConfig
-	db               *gosql.DB
-	singletonDB      *gosql.DB
-	migrationContext *base.MigrationContext
+	connectionConfig  *mysql.ConnectionConfig
+	db                *gosql.DB
+	singletonDB       *gosql.DB
+	migrationContext  *base.MigrationContext
+	finishedMigrating bool
```

Comment:
This variable is accesses read/write by different goroutines. You'd need o change it to `int64` and use `atomic.StoreInt64` and `atomic.LoadInt64` ; see various such examples throughout the code.